src/components/Dashboard/Modals/RenameBoard/RenameBoardModal.tsx

```diff
@@ -0,0 +1,66 @@
+import { ContextModalProps, modals } from '@mantine/modals';
+import { Alert, Button, TextInput } from '@mantine/core';
+import { api } from '~/utils/api';
+import { useForm, zodResolver } from '@mantine/form';
+import { z } from 'zod';
+import { IconAlertCircle } from '@tabler/icons-react';
+import { useTranslation } from 'next-i18next';
+
+type RenameBoardModalProps = {
+  configName: string;
+  configNames: string[];
+}
+
+export const RenameBoardModal = ({ context, id, innerProps }: ContextModalProps<RenameBoardModalProps>) => {
+  const { t } = useTranslation(['manage/boards', 'common']);
+
+  const utils = api.useUtils();
+  const { mutateAsync: mutateRenameBoardAsync, isLoading, isError, error } = api.boards.renameBoard.useMutation({
+    onSettled: () => {
+      void utils.boards.all.invalidate();
+    }
+  });
+
+  const form = useForm({
+    initialValues: {
+      newName: '',
+    },
+    validate: zodResolver(z.object({
+      newName: z.string().min(1).refine(value => !innerProps.configNames.includes(value)),
```

Comment:
Should we force exclude special characters? Since the name is also used to make the file name, it has to follow file naming schemes right?